Staging deploy Github Actions Configs

AratKruglik opened the pull request.

AratKruglik committed:
feat: add GitHub Actions workflow for staging deployment

AratKruglik committed:
feat: update deployment environment to mw-staging

AratKruglik committed:
feat: update CI workflow to use mw-staging environment

AratKruglik committed:
feat: update CI workflow to use ubuntu-latest environment

AratKruglik committed:
feat: update deployment workflow to trigger on push events and streamline environment setup

AratKruglik committed:
Merge branch 'develop' into feature/47-staging-deploy

# Conflicts:
#	.github/workflows/deploy.yml

AratKruglik committed:
feat: update deployment workflow to trigger on manual dispatch instead of push events

AratKruglik committed:
feat(docker): add user mapping to nginx service in docker-compose

AratKruglik committed:
feat: fix log permissions in deploy workflow

AratKruglik committed:
feat: export UID and GID to environment variables in deploy workflow

AratKruglik committed:
refactor: replace inline .env file creation with write-file-action in deploy workflow

AratKruglik committed:
Merge branch 'develop' into feature/47-staging-deploy

AratKruglik committed:
fix: reorder steps in deploy workflow to ensure code is checked out before exporting UID:GID

maks-oleksyuk commented on .github/workflows/deploy.yml:
```diff
@@ -2,50 +2,53 @@ name: Deploy To Staging
 
-          docker compose exec mw-app yarn install --frozen-lockfile
-          docker compose exec mw-app yarn build
+          docker compose exec app yarn install --frozen-lockfile
```
в накидку чи не було б добре додати ще `maintance mode` + `optimize:clear` останнім кроком?

AratKruglik committed:
Merge branch 'develop' into feature/47-staging-deploy

AratKruglik committed:
feat: update deployment workflow for refined service control and reliability

AratKruglik committed:
feat: ensure correct permissions for workspace log directory in deploy workflow

AratKruglik committed:
feat: update deploy workflow and redirect nginx logs to stdout/stderr

AratKruglik committed:
feat: set storage and cache permissions in deploy workflow

AratKruglik committed:
Merge branch 'develop' into feature/47-staging-deploy

maks-oleksyuk commented on .github/workflows/deploy.yml:
```diff
@@ -2,50 +2,117 @@ name: Deploy To Staging
-      branch:
-        description: 'Branch to deploy'
-        required: true
-        default: 'main'
```
чи не треба нам тут обмежити запуск тільки на 1 гілку?

AratKruglik replied:
Файно мати можливість деплоїти будь-яку гілку і мати можливість її тестувати.

maks-oleksyuk replied:
що робити в такому випадку з міграціями? чи не стріляємо ми цим собі в ногу?

AratKruglik replied:
Збагойна. Це все ще в роботі.

maks-oleksyuk commented on .github/workflows/deploy.yml:
```diff
@@ -2,50 +2,117 @@ name: Deploy To Staging
-          cat > .env << 'EOL'
-          ${{ secrets.ENV_FILE }}
-          EOL
+          echo "${{ secrets.ENV_FILE }}" > .env
```
якщо це одне й те саме середовище, а не чисте, чи не буде в такому випадку `.env` збільшуватись при кожному деплої?

AratKruglik replied:
Скоріше за все, а точніше на 100% буде. Тому цей драфт потрбіно доточувати.

AratKruglik replied:
пофікшено

basilstr committed:
User Profile Form (#53)

Co-authored-by: Oleksii Kruhlyk <oleksii.kruhlyk@gmail.com>

zloykotuk committed:
Created a custom error page (#39) (#45)

Co-authored-by: Oleksii Kruhlyk <oleksii.kruhlyk@gmail.com>

AratKruglik committed:
Merge branch 'feature/47-staging-deploy' into feature/35-mentors-program-list-page

AratKruglik committed:
fix: update expertise and clarify mutation testing in coding standards

AratKruglik committed:
feat: add staging deployment configuration with SSL and Docker support

AratKruglik committed:
feat: ensure .env file creation is consistent during build and deployment

AratKruglik committed:
feat: ensure .env file existence during Docker build and deployment

AratKruglik committed:
feat: set working directory earlier in PHP Dockerfile

AratKruglik committed:
feat: optimize Dockerfile for PHP by reordering copy and dependency installation steps

AratKruglik committed:
feat(docker): allow composer files in Docker build context

AratKruglik committed:
feat: adjust Dockerfile to optimize copy sequence and streamline build process

AratKruglik committed:
feat: handle composer install failure gracefully in PHP Dockerfile

AratKruglik committed:
feat: comment out healthcheck in staging docker-compose file

AratKruglik committed:
feat: enable healthcheck for staging with non-blocking test

AratKruglik committed:
feat: specify services to wait for in deployment workflow

AratKruglik committed:
feat: add health check endpoint and improve nginx HTTP to HTTPS handling in staging config

AratKruglik committed:
feat: update deployment workflow and coding guidelines

- Revamped GitHub Actions deploy workflow for better clarity and flexibility.
- Migrate and enhance coding standards documentation.

AratKruglik committed:
feat(workflow): switch SSH setup to shimataro/ssh-key-action for enhanced key handling

AratKruglik committed:
feat: automate known_hosts generation in deploy workflow

AratKruglik committed:
feat: update deploy workflow for staging with Docker, SSL, and health checks

AratKruglik committed:
feat: add Docker image build and deployment workflows with environment-specific config

AratKruglik committed:
feat: update deploy workflow to use self-hosted runner and production Docker Compose

AratKruglik committed:
feat: add production Docker Compose configuration and update staging image reference

AratKruglik committed:
feat: enhance deploy workflow with first image tag extraction and correct compose file usage

AratKruglik committed:
feat: update deployment scripts to use .env file for Docker Compose and set fallback image

AratKruglik committed:
feat: update deployment workflow to use Git SHA for image tagging and improve fallback logic

AratKruglik committed:
feat: simplify deployment workflow by removing input parameters and enhancing Docker image handling

AratKruglik committed:
feat: migrate to FrankenPHP, simplify stack, and optimize Docker configuration

AratKruglik committed:
feat: streamline deployment workflow by enhancing Docker image handling and improving health check logic

AratKruglik committed:
feat: enhance deployment workflow with permission fixes and add frankenphp worker

AratKruglik committed:
feat: update environment configuration to use FrankenPHP as the Octane server
